src/setup.rs

```diff
@@ -93,6 +93,53 @@ pub(crate) fn setup() {
         rcc.ahb1rstr.modify(|_, w| w.ethmacrst().set_bit());
         rcc.ahb1rstr.modify(|_, w| w.ethmacrst().clear_bit());
     });
+
+    #[cfg(feature = "stm32f1xx-hal")]
+    cortex_m::interrupt::free(|_| unsafe {
```

Comment:
I would prefer this to be done with bit banding, but we can leave that for another PR.